packages/async-flow/src/types.d.ts

```diff
@@ -54,10 +55,12 @@ type Simplify<T> = { [KeyType in keyof T]: T[KeyType] } & {};
  */
 type HostInterface<T> = {
   [K in keyof T]: T[K] extends CallableFunction
-    ? HostOf<T[K]>
+    ? HostFn<T[K]>
     : T[K] extends Record<string, any>
       ? Simplify<HostInterface<T[K]>>
-      : T[K];
+      : T[K] extends Passable
+        ? T[K]
```

Comment:
I think we need to detect the `Promise` case here, and for now map it to `never` as well.